clang/utils/generate_unsupported_in_drivermode.py

```diff
@@ -0,0 +1,626 @@
+for option in options_json["!instanceof"]["Option"]:
+    kind = options_json[option]["Kind"]["def"]
+    tmp_visibility_list = []
+    group_sequence = []
+    option_name = options_json[option]["Name"]
+
+    # There are a few conditions that make an option unsuitable to test in this script
+    # Options of kind KIND_INPUT & KIND_UNKNOWN don't apply to this test. For example,
+    # Option "INPUT" with name "<input>".
+    if (
+        option_name in exceptions_sequence
+        or options_json[option]["Name"] is None
+        or kind == "KIND_INPUT"
+        or kind == "KIND_UNKNOWN"
+    ):
+        untested_sequence.append(
+            UnsupportedDriverOption("All", option, option_name, "")
+        )
+        continue
+
+    # Get the correct option prefix
+    prefixes = options_json[option]["Prefixes"]
+    prefix = ""
+    if prefixes is not None and len(prefixes) > 0:
+        # Assuming the first prefix is the preferred prefix
+        prefix = prefixes[0]
+
+    get_visibility(option, tmp_visibility_list)
+
+    # Check visibility of direct and indirect aliases
+    # A given option may list only one "primary" alias, but that alias
+    # may be listed by other options as well, hence indirect aliases
+    alias_sequence = options_json["!instanceof"]["Alias"]
+
+    if options_json[option]["Alias"] is not None:
+        primary_alias = options_json[option]["Alias"]["def"]
+
+        get_visibility(primary_alias, tmp_visibility_list)
+
+        for alias in alias_sequence:
+            if options_json[alias]["Alias"]["def"] == primary_alias:
+                get_visibility(alias, tmp_visibility_list)
+
+    for alias in alias_sequence:
+        if options_json[alias]["Alias"]["def"] == option:
+            get_visibility(alias, tmp_visibility_list)
+
+    # KIND_JOINED* options that are supported need to be saved for checking
+    # which options cannot be validated with this script
+    is_option_kind_joined = kind == "KIND_JOINED" or kind == "KIND_JOINED_OR_SEPARATE"
+
+    # Append to the unsupported list, and the various supported lists
+    for driver in driver_sequence:
+        if driver not in tmp_visibility_list:
+            unsupported_sequence.append(
+                UnsupportedDriverOption(driver, option, option_name, prefix)
+            )
+        elif is_option_kind_joined:
+            driver_controller[get_index(driver)].supported_sequence.append(
+                UnsupportedDriverOption(driver, option, option_name, prefix)
+            )
+
+# Sort the supported lists for the next block
+for driver_ctrl in driver_controller:
+    driver_ctrl.supported_sequence.sort(key=len, reverse=True)
+
+# For a given driver, this script should not generate tests for unsupported options
+# whose option Name have a prefix that corresponds to a supported option / visible option of Kind KIND_JOINED*.
+# These driver-option pairs are removed here.
```

Comment:
> For a given driver, this script **should** not generate tests for unsupported options
whose option Name have a prefix that corresponds to a supported option / visible option of Kind KIND_JOINED*.

Its not that it should not, instead it cannot, because that will be parsed as the prefixed option with a value and thus will not cause an error. Would be good to add a comment explaining this with a specific example.